src/uct/tcp/tcp_ep.c

```diff
@@ -556,10 +610,12 @@ void uct_tcp_ep_pending_purge(uct_ep_h tl_ep, uct_pending_purge_callback_t cb,
 ucs_status_t uct_tcp_ep_flush(uct_ep_h tl_ep, unsigned flags,
                               uct_completion_t *comp)
 {
-    uct_tcp_ep_t *ep = ucs_derived_of(tl_ep, uct_tcp_ep_t);
+    uct_tcp_ep_t *ep    = ucs_derived_of(tl_ep, uct_tcp_ep_t);
+    ucs_status_t status = uct_tcp_ep_can_send(ep);
 
-    if (!uct_tcp_ep_can_send(ep)) {
-        return UCS_ERR_NO_RESOURCE;
+    if (ucs_unlikely(status != UCS_OK)) {
```

Comment:
what if the status is UNREACHABLE because the connection is closed? shouldn't we return UCS_OK?